### PR TITLE
fix timestep and learning rate recording

### PR DESCRIPTION
**Pull request by sc420**
Hi,
I've found two bugs in A2C training code and fixed them:

1. How number of timesteps is updated (was updated with an extra +1)
2. How learning rate is recorded (it should use placeholder)

The bugs would lead to:

1. If we train for 10M timesteps, we would see the data to have than 10M timesteps in tensorboard at the end
2. If we use linear type of annealing, we would see the initial learning rate in tensorboard all the time

It seems to only affect how the training is logged in tensorboard, it doesn't have any effects on training.

### CODE REVIEW

File: stable_baselines/a2c/a2c.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def learn(self, total_timesteps, callback=None, seed=None, log_interval=100, tb_
                 obs, states, rewards, masks, actions, values, ep_infos, true_reward = runner.run()
                 ep_info_buf.extend(ep_infos)
                 _, value_loss, policy_entropy = self._train_step(obs, states, rewards, masks, actions, values,
-                                                                 self.num_timesteps // (self.n_batch + 1), writer)
+                                                                 update - 1, writer)
```
Comment (araffin) [inline stable_baselines/a2c/a2c.py]: If you don't use `self.num_timesteps`, then it will be wrong when you call `model.learn()` with `reset_num_timesteps=False`.

Comment (sc420) [reply stable_baselines/a2c/a2c.py]: Thanks, I didn't notice that. I've fixed it in new commits.

File: stable_baselines/a2c/a2c.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def learn(self, total_timesteps, callback=None, seed=None, log_interval=100, tb_
                                                                       masks.reshape((self.n_envs, self.n_steps)),
                                                                       writer, self.num_timesteps)
 
-                self.num_timesteps += self.n_batch + 1
+                self.num_timesteps += self.n_batch
```
Comment (araffin) [inline stable_baselines/a2c/a2c.py]: thinking again about it, this line should be right after `runner.run()` no?
Otherwise, the first update will be `step=0` in tensorboard

Comment (sc420) [reply stable_baselines/a2c/a2c.py]: `_train_step` takes place before this line of update, so the first update will be `step=0` no matter what this line (:252) does. The example of logging the first step to be 0 can be seen at this [TensorFlow example](https://github.com/aymericdamien/TensorFlow-Examples/blob/master/examples/4_Utils/tensorboard_basic.py#L82).

Comment (araffin) [reply stable_baselines/a2c/a2c.py]: >takes place before this line of update

yes, that's what I'm proposing, moving the line before train_step call. Except if you think that's ok to keep it like that ;)

Comment (sc420) [reply stable_baselines/a2c/a2c.py]: I think it's just a matter of preference, but I would prefer the original version because it's easier to process the log as `for update_index in range(num_updates)` and it's consistent with the approaches in other algorithms. If we move the line, we should also consider moving `total_episode_reward_logger`, otherwise the timesteps logged in `episode_reward` would also be incorrectly offset.
